File: python/Prodi.py
# import class yang saling berhubungan (Composite)
from Mahasiswa import Mahasiswa
from Dosen import Dosen
from Course import Course

class Prodi:
    # deklarasi property private
    __nama_prodi = ""
    __kode = ""
    __course = []
    __mahasiswa = []
    __dosen = []

    # membuat konstruktor
    def __init__(self, namaprodi, kode, Mahasiswa = [], Dosen = [], Course = []):
        self.__nama_prodi = namaprodi
        self.__kode = kode
        self.__mahasiswa = Mahasiswa
        self.__dosen = Dosen
        self.__course = Course
    
    # membuat getter
    def getNamaProdi(self):
        return self.__nama_prodi
    
    def getKode(self):
        return self.__kode
    
    def getMahasiswa(self):
        for i, data in enumerate(self.__mahasiswa):
            print("Data Mahasiswa ", i+1)
            print("NIK :", data.getNik())
            print("Nama :", data.getNama())
            print("Jenis Kelamin :", data.getGender())
            print("Asal Universitas :", data.getAsalUniversitas())
            print("Email :", data.getEmail())
            print("NIM :", data.getNim())
            print("Fakultas :", data.getFakultas())
            print("\n")

    def getDosen(self):
        for i, data in enumerate(self.__dosen):
            print("Data Dosen ", i+1)
            print("NIK :", data.getNik())
            print("Nama :", data.getNama())
            print("Jenis Kelamin :", data.getGender())
            print("Asal Universitas :", data.getAsalUniversitas())
            print("Email :", data.getEmail())
            print("NIP :", data.getNip())
            print("Fakultas :", data.getFakultas())
            print("Pendidikan Terakhir :", data.getPendterakhir())
            print("Keahlian :", data.getKeahlian())
            print("\n")

    def getCourse(self):
        for i, data in enumerate(self.__course):
            print("Data Course ", i+1)
            print("Mata Kuliah :", data.getNamamatkul())
        print("\n")

File: c++/Prodi.cpp
#include <iostream>
#include <list>
#include <string>
using namespace std;

#include "Mahasiswa.cpp"
#include "Dosen.cpp"
#include "Course.cpp"

class Prodi{
    private:
        string namaprodi;
        string kode;
        list<Course> data2;
        list<Mahasiswa> data;
        list<Dosen> data1;

    public:
        Prodi(){

        }

        Prodi(string namaprodi = "", string kode = "", list<Mahasiswa> data = {}, list<Dosen> data1 = {}, list<Course> data2 = {}){
            this->namaprodi = namaprodi;
            this->kode = kode;
            this->data = data;
            this->data1 = data1;
            this->data2 = data2;
        }

        string getNamaProdi(){
            return this->namaprodi;
        }

        string getKode(){
            return this->kode;
        }

        void getMahasiswa(){
            int i = 0;
            for(Mahasiswa cari:data){
                cout << "Data Mahasiswa " << ++i << endl;
                cout << "\tNIK : " << cari.getNik() << endl;
                cout << "\tNama : " << cari.getNama() << endl;
                cout << "\tJenis Kelamin : " << cari.getGender() << endl;
                cout << "\tAsal Universitas : " << cari.getAsalUniversitas() << endl;
                cout << "\tEmail : " << cari.getEmail() << endl;
                cout << "\tNIM : " << cari.getNim() << endl;
                cout << "\tFakultas : " << cari.getFakultas() << endl;
                cout << endl;
            }
        }

        void getDosen(){
            int i = 0;
            for(Dosen cari:data1){
                cout << "Data Dosen " << ++i << endl;
                cout << "\tNIK : " << cari.getNik() << endl;
                cout << "\tNama : " << cari.getNama() << endl;
                cout << "\tJenis Kelamin : " << cari.getGender() << endl;
                cout << "\tAsal Universitas : " << cari.getAsalUniversitas() << endl;
                cout << "\tEmail : " << cari.getEmail() << endl;
                cout << "\tNIP : " << cari.getNip() << endl;
                cout << "\tFakultas : " << cari.getFakultas() << endl;
                cout << "\tPendidikan Terakhir : " << cari.getPendTerakhir() << endl;
                cout << "\tKeahlian : " << cari.getKeahlian() << endl;
                cout << endl;
            }
        }

        void getCourse(){
            int i = 0;
            for(Course cari:data2){
                cout << "Data Course " << ++i << endl;
                cout << "\tMata Kuliah : " << cari.getNamaMatkul() << endl;
            }
        }
        
        void setNamaProdi(string namaprodi){
            this->namaprodi = namaprodi;
        }

        void setKode(string kode){
            this->kode = kode;
        }

        ~Prodi(){

        }
};